tensorbay/label/tests/test_subcatalog.py

```diff
@@ -0,0 +1,203 @@
+_COMMON_SUBCATALOG_CONTENT = dict(_DATA_TRACKING, **_DATA_CATEGORIES, **_DATA_ATTRIBUTES)
+_CLASSIFICATION_SUBCATALOG = dict(_DATA_CATEGORIES, **_DATA_ATTRIBUTES, **_DATA_DELIMITER)
+_KEYPOINTS2D_SUBCATALOG = dict(
+    _DATA_TRACKING, **_DATA_CATEGORIES, **_DATA_ATTRIBUTES, **_DATA_KEYPOINTS
+)
+_SENTENCE_SUBCATALOG = dict(_DATA_ATTRIBUTES, **_DATA_SENTENCE)
+
+
+@pytest.fixture
+def categories():
+    categories = NameOrderedDict()
+    for name in _DATA_CATEGORIES["categories"]:
+        categories.append(CategoryInfo(name["name"]))
+    return categories
+
+
+@pytest.fixture
+def attributes():
+    attributes = NameOrderedDict()
+    attributes.append(AttributeInfo("gender", enum=["male", "female"]))
+    return attributes
+
+
+@pytest.fixture
+def keypoints():
+    return KeypointsInfo(
+        5,
+        names=_DATA_NAMES,
+        skeleton=_DATA_SKELETON,
+    )
+
+
+class TestClassificationSubcatalog:
+    def test_init_subclass(self):
+        classification_subcatalog = ClassificationSubcatalog()
+        classification_subcatalog._supports = (SupportCategories, SupportAttributes)
+
+    def test_loads(self):
+        classification_subcatalog = ClassificationSubcatalog.loads(_CLASSIFICATION_SUBCATALOG)
+        assert list(classification_subcatalog.categories) == _LIST_CATAGORIES
+        assert classification_subcatalog.attributes["gender"].enum == ["male", "female"]
+
+    def test_dumps(self, categories, attributes):
+        classification_subcatalog = ClassificationSubcatalog()
+        classification_subcatalog.category_delimiter = "-"
+        classification_subcatalog.categories = categories
+        classification_subcatalog.attributes = attributes
+        assert classification_subcatalog.dumps() == _CLASSIFICATION_SUBCATALOG
+
+
+class TestCommnoSubcatalog:
+    """This class used to test multiple classes.
+
+    Box2DSubcatalog, Box3DSubcatalog,
+    Polygon2DSubcatalog, Polyline2DSubcatalog.
+    """
+
+    @pytest.mark.parametrize(
+        "COMMON_SUBCATALOG",
+        [Box2DSubcatalog, Box3DSubcatalog, Polygon2DSubcatalog, Polyline2DSubcatalog],
+    )
+    def test_init_subclass(self, COMMON_SUBCATALOG):
+        subcatalog = COMMON_SUBCATALOG()
+        assert subcatalog._supports == (
+            SupportIsTracking,
+            SupportCategories,
+            SupportAttributes,
+        )
+
+    @pytest.mark.parametrize(
+        "COMMON_SUBCATALOG",
+        [Box2DSubcatalog, Box3DSubcatalog, Polygon2DSubcatalog, Polyline2DSubcatalog],
+    )
+    def test_loads(self, COMMON_SUBCATALOG):
+        subcatalog = COMMON_SUBCATALOG.loads(_COMMON_SUBCATALOG_CONTENT)
+
+        assert subcatalog.is_tracking == True
+        assert list(subcatalog.categories) == _LIST_CATAGORIES
+        assert subcatalog.attributes["gender"].enum == ["male", "female"]
+
+    @pytest.mark.parametrize(
+        "COMMON_SUBCATALOG",
+        [Box2DSubcatalog, Box3DSubcatalog, Polygon2DSubcatalog, Polyline2DSubcatalog],
+    )
```

Comment:
decorator is a function object, it can be a variable:
```python
parametrize = pytest.mark.parametrize(
     "COMMON_SUBCATALOG",
     [Box2DSubcatalog, Box3DSubcatalog, Polygon2DSubcatalog, Polyline2DSubcatalog],
)

@paramtrize
def test_xxx():
```

Reply:
Also `tuple` is better than `list`